Sources/NIOTransportServices/NIOFilterEmptyWritesHandler.swift

```diff
@@ -88,9 +88,16 @@ public final class NIOFilterEmptyWritesHandler: ChannelDuplexHandler {
 // Connection state management
 extension NIOFilterEmptyWritesHandler {
     public func channelActive(context: ChannelHandlerContext) {
-        assert(self.state == .notActiveYet)
-        self.state = .open
-        context.fireChannelActive()
+        switch self.state {
+        case .notActiveYet:
+            self.state = .open
+            context.fireChannelActive()
+        case .closedFromLocal:
+            // Closed before we activated, not a problem.
+            assert(self.prefixEmptyWritePromise == nil)
+        case .open, .closedFromRemote, .error:
+            preconditionFailure()
```

Comment:
Not worth having a tiny little message here?

Reply:
I think it's largely unnecessary: on iOS crash reports will point directly at this line anyway, and the static string raises binary sizes with minimal improvement in debuggability.